securedrop_client/logic.py

```diff
@@ -99,7 +96,11 @@ def __init__(self, hostname, gui, session, home: str) -> None:
         self.api = None  # Reference to the API for secure drop proxy.
         self.session = session  # Reference to the SqlAlchemy session.
         self.api_thread = None  # Currently active API call thread.
+        self.home = home # used for finding DB in sync thread
         self.sync_flag = os.path.join(home, 'sync_flag')
+        self.home = home  # The "home" directory for client files.
+        self.data_dir = os.path.join(self.home, 'data')  # File data.
+        self.timer = None # call timeout timer
```

Comment:
The `Client` object now has a timer, to detect API calls which take too long. The `home` stuff above is unrelated to the threading changes, I'll remove it.